### DISCUSSION TITLE
Bug/Enhancement Filter WebPart

### DISCUSSION THREAD

**Discussion by myatix** [microsoft-search/pnp-modern-search#1516] [Ideas]
**Version used**
Ex: 4.4.1

**Describe the bug**
If you configure the refiners in the filter webpart to only show a certain number of items "# of values" and there are actually more items than the set "# of values" to available, no scrollbar appears showing that there are more refiner items.

Would it not be a good idea that the refiner scrolled to show the additional items when you set a "# of values"? Just a thought but you could easily end up in a situation where you have a larger number of items (EG:100) in the refiner and don't want to scroll the whole page to find the refiner you want to use. Then have to scroll all the way to the top again to view the results.

**To Reproduce**
1. Configure the Refiner to use a large number refiner values
2. Limit the # of values to EG: 5

![image](https://user-images.githubusercontent.com/48433526/140326801-d7b39910-849e-4564-b627-b2f57a6ce77c.png)

Perhaps another solution to the problem could also be a good idea but it's not a great UX if a user needs to scroll to the bottom of a long page to pick a refiner and then scroll all the way to the top to see the results.

**Comment by wobba** [comment]
The number is an API limit, not a UX limit. Restricting in the UX should be in the template.

**Comment by myatix** [comment]
@wobba I don’t think you have understood what I am saying! It’s nothing to do with limits! It’s the way the UI shows really long lists of refinable items.

**Comment by wobba** [comment]
My question, how can you configure to get a large set of filter values, when you set it to 5? The setting "# of values" is an API restriction today - not a UX restriction. Your point 1 and 2 contradict each other the way I see it.

**Comment by myatix** [comment]
Perhaps I have miss understood the use of “# of values”, but my understanding is that # of values is designed to limit the number of viewable values returned???

That’s great! But what if you want to limit the number of viewable values but then have the possibility of scrolling in the filter webpart to see the other available values. Rather than just having a huge list. Which can cause usability issues if the list is too long.

I will try and put together a wireframe when I have time if this still doesn’t make sense and perhaps take some screenshots.

I am talking about the UI and not the limiting factors in the api.

**Comment by wobba** [comment]
As I said "# of values" is how many values are retrieved from the API in case you want to limit/raise the default which is 100.

If you want to limit in UX you should look into a custom filter template.

**Comment by myatix** [comment]
@wobba Thanks for the response! :)

OK... In that case I am not using "# of values" as intended.

Is there any other way to limit the refiner webpart so it doesn't show 100's of refiners in a huge list but instead provides maybe a scroll function or something similar?

**Comment by wobba** [comment]
You need a custom filter template. I'll move this over to an enhancement.
